src/account.ts

```diff
@@ -301,82 +295,86 @@ export default class EthereumAccount {
-      ]
-
-      // Run each settle task sequentially, spending down from the budget
-      amountLeftover = await tasks.reduce(async (leftover: Promise<BigNumber>, task: SettleTask) => {
-        const budget = await leftover
-        return budget.lte(0) ? budget : task(budget)
-      }, Promise.resolve(settlementBudget))
-    } catch (err) {
-      this.master._log.error(`Error during settlement: ${err.message}`)
-    } finally {
-      const amountSettled = settlementBudget.minus(amountLeftover)
-
-      if (amountSettled.gt(0)) {
-        this.master._log.trace(`Settle attempt complete: spent total of ${format(amountSettled, Unit.Wei)} settling, ` +
-          `refunding ${format(amountLeftover, Unit.Wei)} back to balance with ${this.account.accountName}`)
-      } else if (amountSettled.isZero()) {
-        this.master._log.trace(`Settle attempt complete: none of budget spent, ` +
-          `refunding ${format(amountLeftover, Unit.Wei)} back to balance with ${this.account.accountName}`)
-      } else {
-        this.master._log.error(`Critical settlement error: spent ${format(amountSettled, Unit.Wei)}, ` +
-          `more than budget of ${format(settlementBudget, Unit.Wei)} with ${this.account.accountName}`)
-      }
+      try {
+        this.master._log.info(`Settlement triggered with account ${this.account.accountName} for maximum of ${format(settlementBudget, Unit.Gwei)}`)
+
+        // This should never error, since settleTo < maximum
+        this.addBalance(settlementBudget)
+        this.account.payoutAmount = this.account.payoutAmount.minus(settlementBudget)
+
+        // Convert gwei to wei
+        settlementBudget = convert(settlementBudget, Unit.Gwei, Unit.Wei).dp(0, BigNumber.ROUND_FLOOR)
+
+        type SettleTask = (budget: BigNumber) => Promise<BigNumber>
+        const tasks: SettleTask[] = [
+          // 1) Try to send a claim: spend the channel down entirely before funding it
+          b => this.sendClaim(b),
+          // 2) Open or deposit to a channel if it's necessary to send anything leftover
+          b => this.fundOutgoingChannel(b),
+          // 3) Try to send a claim again since we may have more funds in the channel
+          b => this.sendClaim(b)
+        ]
+
+        // Run each settle task sequentially, spending down from the budget
+        amountLeftover = await tasks.reduce(async (leftover: Promise<BigNumber>, task: SettleTask) => {
+          const budget = await leftover
+          return budget.lte(0) ? budget : task(budget)
+        }, Promise.resolve(settlementBudget))
+      } catch (err) {
+        this.master._log.error(`Error during settlement: ${err.message}`)
+      } finally {
+        const amountSettled = settlementBudget.minus(amountLeftover)
+
+        const spent = format(amountSettled, Unit.Wei)
+        const leftover = format(amountLeftover, Unit.Wei)
+        const account = this.account.accountName
+
+        if (amountSettled.eq(settlementBudget)) {
```

Comment:
It would be helpful to explain with comments why we might end up with each of these results